Errors with Caddy 2.2.1

I'm compiling Caddy 2.2.1 with xcaddy using Go 1.14.4 on Ubuntu 20.04.1 LTS

```
xcaddy build \
    --with github.com/abiosoft/caddy-json-schema \
    --with github.com/caddy-dns/cloudflare
```
Caddy builds and runs fine, executing `caddy json-schema --vscode` gives me this message

```
panic: reflect: call of reflect.Value.Type on zero Value
 
goroutine 1 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0xc0006d7130, 0xc00073ee78)
        reflect/value.go:1872 +0x183
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc0008750e0, 0x0, 0x0)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:143 +0xef
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc000875050, 0x148df40, 0xc0005c5860)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:132 +0x38e
github.com/abiosoft/caddy-json-schema.(*Interface).populateStruct(0xc000874e10, 0x1a7a5a0, 0x15fbd00)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:186 +0xd4a
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc000874e10, 0x15fbd00, 0xc0005305c0)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:120 +0x2ac
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc000874d80, 0x1488a80, 0xc0005c5840)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:132 +0x38e
github.com/abiosoft/caddy-json-schema.(*Interface).populateStruct(0xc00073fc88, 0x1a7a5a0, 0x1634e60)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:186 +0xd4a
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc00073fc88, 0x1634e60, 0xc0004142a0)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:120 +0x2ac
github.com/abiosoft/caddy-json-schema.(*Interface).populate(0xc00073fc88, 0x15a28c0, 0xc000115bc0)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/interface.go:124 +0x14e
github.com/abiosoft/caddy-json-schema.generateSchema(0x0, 0x0)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/schema_util.go:51 +0x159
github.com/abiosoft/caddy-json-schema.run(0xc000115a40, 0xc00003a1a0, 0x1, 0x1)
        github.com/abiosoft/caddy-json-schema@v0.0.0-20200527180432-2d0cb96ed8ea/command.go:67 +0x3d
github.com/caddyserver/caddy/v2/cmd.Main()
        github.com/caddyserver/caddy/v2@v2.2.1/cmd/main.go:85 +0x239
main.main()
        caddy/main.go:13 +0x20
```

--- a/interface.go
+++ b/interface.go
@@ -16,9 +16,10 @@ type Interface struct {
 	Module string
 
 	// properties
-	Name   string
-	Fields []Interface
-	Type   string
+	Name     string
+	Fields   []Interface
+	Type     string
+	Nullable bool
 
 	// array/map type
 	Array bool
@@ -54,6 +55,7 @@ func (f Interface) goPkg() string {
 func (f Interface) toSchema() *Schema {
 	var s = NewSchema()
 	s.setType(f.Type)
+	s.Nullable = f.Nullable
 
 	// if it's a module loader, construct a special case (sub)schema
 	if len(f.Loader) > 0 {
@@ -77,6 +79,7 @@ func (f Interface) toSchema() *Schema {
 			cs.setType("array")
 			cs.ArrayItems = NewSchema()
 			cs.ArrayItems.setType(nest.Type)
+			cs.ArrayItems.Nullable = nest.Nullable
 			cs.ArrayItems.Properties = props
 
 			// nested schema
@@ -111,6 +114,14 @@ func (f Interface) toSchema() *Schema {
 
 // populate populates the Interface with the type of s.
 func (f *Interface) populate(s interface{}) {
+
+	if s == nil {
+		// interface{} (s == <nil>) is marshalled to ["string", "null"] for the time being
+		f.Type = "string"
+		f.Nullable = true
+		return
+	}
+
 	v := reflect.ValueOf(s)
 
 	elemVal := func() interface{} { return reflect.Zero(v.Type().Elem()).Interface() }

--- a/schema.go
+++ b/schema.go
@@ -1,6 +1,7 @@
 package jsonschema
 
 import (
+	"encoding/json"
 	"fmt"
 	"strings"
 )
@@ -26,6 +27,7 @@ type Schema struct {
 	Description         string `json:"description,omitempty"`
 	MarkdownDescription string `json:"markdownDescription,omitempty"`
 	Type                string `json:"type,omitempty"`
+	Nullable            bool   `json:"-"`
 	Ref                 string `json:"$ref,omitempty"`
 
 	ArrayItems           *Schema            `json:"items,omitempty"`
@@ -117,4 +119,19 @@ func (s *Schema) setType(typ string) {
 
 func (s *Schema) setRef(moduleID string) {
 	s.Ref = "#/definitions/" + moduleID
+}
+
+// MarshalJSON allows to marshal Schema.Type as string or list
+func (s *Schema) MarshalJSON() ([]byte, error) {
+	type Alias Schema
+	if s.Nullable {
+		return json.Marshal(&struct {
+			Type [2]string `json:"type"`
+			*Alias
+		}{
+			Type:  [2]string{s.Type, "null"},
+			Alias: (*Alias)(s),
+		})
+	}
+	return json.Marshal(&struct{ *Alias }{Alias: (*Alias)(s)})
 }

